### PR TITLE
bank: Prune older epoch stakes

### LINKED ISSUE #6991: bank.epoch_stakes grows without bound
#### Problem
the bank never culls epoch stakes

#### Proposed Solution
retain at most N epochs worth of epoch_stakes
```
if leader_schedule_epoch > MAX_LEADER_SCHEDULE_STAKES {
    new.epoch_stakes.remove(leader_schedule_epoch - MAX_LEADER_SCHEDULE_STAKES)
}
```
in bank.new_from_parent()

### PR DESCRIPTION
**Pull request by mvines**
Fixes #6991

### CODE REVIEW

**Comment by mvines** [comment]
crud, `ci/test-local-cluster.sh` passes fine locally with this patch

**Comment by mvines** [comment]
🐣

**Comment by mvines** [comment]
Thanks @ryoqun, I pulled in most of #7098.  I found no need for your changes in `ledger/src/leader_schedule_cache.rs` so I dropped those hunks

File: runtime/src/bank.rs [hunk 1]
```diff
@@ -60,6 +60,8 @@ use std::{
 
 pub const SECONDS_PER_YEAR: f64 = (365.25 * 24.0 * 60.0 * 60.0);
 
+pub const MAX_LEADER_SCHEDULE_STAKES: Epoch = 5;
```
Comment (mvines) [inline runtime/src/bank.rs]: 5?  🤷‍♂ - I have no reason for 5 over any other number at the moment

**Comment by mvines** [comment]
@rob-solana or @sagar-solana - would you mind quickly scanning this PR?   It's pretty close to @ryoqun's original work

**Comment by mvines** [comment]
I'm going to land this to get some test mileage on it before TdS DR6 on Tuesday.  Post-merge review very welcome!

**Comment by ryoqun** [comment]
LGTM

**Comment by rob-solana** [comment]
the only thing holding up @ryoqun's PR was a disposition for absent epoch stakes via RPC.  if that's covered now, all good.

**Pull request merged**